scipy/signal/tests/test_fir_filter_design.py

```diff
@@ -129,6 +129,22 @@ def test_fs_validation(self):
         with pytest.raises(ValueError, match="Sampling.*single scalar"):
             firwin(51, .5, fs=np.array([10, 20]))
 
+    @pytest.mark.parametrize("numtaps, cutoff, pass_zero, expected", [
+        (3, 0.1, False, 0),
+        (3, 0.1, True, 1),
+        (7, 0.1, False, 0),
+        (7, 0.1, True, 1),
+        (1001, 0.1, False, 0),
+    ])
+    def test_gh_19291(self, numtaps, cutoff, pass_zero, expected):
+        actual = firwin(numtaps, cutoff, pass_zero=pass_zero)
+        assert_allclose(actual.sum(), expected, atol=1e-15)
+        if not pass_zero:
+            # manual check via lowpass filter swap
+            kernel = -firwin(numtaps, cutoff)
+            kernel[int(kernel.size / 2)] += 1
+            assert_allclose(actual, kernel)
```

Comment:
@dechamps the first check above passes (coefficients forced to sum to `0` now) as does the rest of the SciPy testsuite, but the coefficients reconstituted from the lowpass->highpass filter conversion don't match perfectly here.

This last check is actually stricter than any checks you made in the original ticket--is it a sign of a problem or a normal/acceptable deviation?